packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -33,13 +33,16 @@ export const getPostDescription = (post: PostsWithNavigation | PostsWithNavigati
       .reduce((acc, curr, i) => {
         const concat = `${acc}${i > 0 ? ` • ` : ""}${curr}`;
         if (acc.length < 40) return concat;
-        if (concat.length < 150) return concat;
+        // If we have room, concatenate the next paragraph. The statelessness of
+        // this reducer makes it hard to tell if this is the next paragraph, so
+        // we cut it off at the second paragraph.
+        if (concat.length < 150 && i <= 1) return concat;
```

Comment:
This feels too hacky to me. If I'm understanding correctly, the purpose is to make sure we don't skip paragraphs when building the post description. I think it would be clearer to just use a for loop and break once you hit a paragraph that would make it too long.

Reply:
Fair. @Xodarap when you see the commit that takes us away from a nice series of chained functional calls and introduces a for loop, you can blame Sarah not me. (JK) I all seriousness, if you know how to make it better in a functional paradigm, I'd be interested to hear about it (I didn't think about it very hard).